### ISSUE TITLE
sqlboiler v4.5.0 generation fails for table containing uuid pk

### ISSUE THREAD

**Issue by cad** [volatiletech/sqlboiler#935]
If you're having a generation problem please answer these questions before submitting your issue. Thanks!

### What version of SQLBoiler are you using (`sqlboiler --version`)?
4.5.0

### What is your database and version (eg. Postgresql 10)
postgresql 13

### If this happened at generation time what was the full SQLBoiler command you used to generate your models? (if not applicable leave blank)
sqlboiler psql --wipe --no-tests -o gen/models

### What is the output of the command above with the `-d` flag added to it? (Provided you are comfortable sharing this, it contains a blueprint of your schema)
```
sqlboiler psql --wipe --no-tests -o gen/models -d
using driver: /Users/cad/go/bin/sqlboiler-psql
unable to fetch table column info (bar): unable to scan for table bar: sql: Scan error on column index 1, name "column_type": converting NULL to string is unsupported
{"config":{"driver_name":"psql","driver_config":{"blacklist":null,"dbname":"<redacted>","host":"localhost","pass":"postgres","port":5432,"schema":"<redacted>","sslmode":"disable","user":"postgres","whitelist":null},"pkg_name":"models","out_folder":"gen/models","debug":true,"no_tests":true,"wipe":true,"struct_tag_casing":"snake","relation_tag":"-","imports":{"all":{"Standard":["\"database/sql\"","\"fmt\"","\"reflect\"","\"strings\"","\"sync\"","\"time\""],"ThirdParty":["\"github.com/friendsofgo/errors\"","\"github.com/volatiletech/sqlboiler/v4/boil\"","\"github.com/volatiletech/sqlboiler/v4/queries\"","\"github.com/volatiletech/sqlboiler/v4/queries/qm\"","\"github.com/volatiletech/sqlboiler/v4/queries/qmhelper\"","\"github.com/volatiletech/strmangle\""]},"test":{"Standard":["\"bytes\"","\"reflect\"","\"testing\""],"ThirdParty":["\"github.com/volatiletech/sqlboiler/v4/boil\"","\"github.com/volatiletech/sqlboiler/v4/queries\"","\"github.com/volatiletech/randomize\"","\"github.com/volatiletech/strmangle\""]},"singleton":{"boil_queries":{"Standard":null,"ThirdParty":["\"github.com/volatiletech/sqlboiler/v4/drivers\"","\"github.com/volatiletech/sqlboiler/v4/queries\"","\"github.com/volatiletech/sqlboiler/v4/queries/qm\""]},"boil_types":{"Standard":["\"strconv\""],"ThirdParty":["\"github.com/friendsofgo/errors\"","\"github.com/volatiletech/sqlboiler/v4/boil\"","\"github.com/volatiletech/strmangle\""]}},"test_singleton":{"boil_main_test":{"Standard":["\"database/sql\"","\"flag\"","\"fmt\"","\"math/rand\"","\"os\"","\"path/filepath\"","\"strings\"","\"testing\"","\"time\""],"ThirdParty":["\"github.com/spf13/viper\"","\"github.com/volatiletech/sqlboiler/v4/boil\""]},"boil_queries_test":{"Standard":["\"bytes\"","\"fmt\"","\"io\"","\"io/ioutil\"","\"math/rand\"","\"regexp\""],"ThirdParty":["\"github.com/volatiletech/sqlboiler/v4/boil\""]},"boil_suites_test":{"Standard":["\"testing\""],"ThirdParty":null}}},"aliases":{},"version":"4.5.0"},"driver_config":{"blacklist":null,"dbname":"<redacted>","host":"localhost","pass":"postgres","port":5432,"schema":"<redacted>","sslmode":"disable","user":"postgres","whitelist":null},"schema":"","dialect":{"lq":0,"rq":0,"use_index_placeholders":false,"use_last_insert_id":false,"use_schema":false,"use_default_keyword":false,"use_auto_columns":false,"use_top_clause":false,"use_output_clause":false,"use_case_when_exists_clause":false},"tables":null,"templates":null}
Error: exit status 1
driver (/Users/cad/go/bin/sqlboiler-psql) exited non-zero
github.com/volatiletech/sqlboiler/v4/drivers.execute
        /Users/cad/go/pkg/mod/github.com/volatiletech/sqlboiler/v4@v4.5.0/drivers/binary_driver.go:73
github.com/volatiletech/sqlboiler/v4/drivers.binaryDriver.Assemble
        /Users/cad/go/pkg/mod/github.com/volatiletech/sqlboiler/v4@v4.5.0/drivers/binary_driver.go:21
github.com/volatiletech/sqlboiler/v4/boilingcore.(*State).initDBInfo
        /Users/cad/go/pkg/mod/github.com/volatiletech/sqlboiler/v4@v4.5.0/boilingcore/boilingcore.go:382
github.com/volatiletech/sqlboiler/v4/boilingcore.New
        /Users/cad/go/pkg/mod/github.com/volatiletech/sqlboiler/v4@v4.5.0/boilingcore/boilingcore.go:93
main.preRun
        /Users/cad/go/pkg/mod/github.com/volatiletech/sqlboiler/v4@v4.5.0/main.go:215
github.com/spf13/cobra.(*Command).execute
        /Users/cad/go/pkg/mod/github.com/spf13/cobra@v1.0.0/command.go:831
github.com/spf13/cobra.(*Command).ExecuteC
        /Users/cad/go/pkg/mod/github.com/spf13/cobra@v1.0.0/command.go:950
github.com/spf13/cobra.(*Command).Execute
        /Users/cad/go/pkg/mod/github.com/spf13/cobra@v1.0.0/command.go:887
main.main
        /Users/cad/go/pkg/mod/github.com/volatiletech/sqlboiler/v4@v4.5.0/main.go:122
runtime.main
        /opt/homebrew/Cellar/go/1.16/libexec/src/runtime/proc.go:225
runtime.goexit
        /opt/homebrew/Cellar/go/1.16/libexec/src/runtime/asm_arm64.s:1130
unable to fetch table data
github.com/volatiletech/sqlboiler/v4/boilingcore.(*State).initDBInfo
        /Users/cad/go/pkg/mod/github.com/volatiletech/sqlboiler/v4@v4.5.0/boilingcore/boilingcore.go:384
github.com/volatiletech/sqlboiler/v4/boilingcore.New
        /Users/cad/go/pkg/mod/github.com/volatiletech/sqlboiler/v4@v4.5.0/boilingcore/boilingcore.go:93
main.preRun
        /Users/cad/go/pkg/mod/github.com/volatiletech/sqlboiler/v4@v4.5.0/main.go:215
github.com/spf13/cobra.(*Command).execute
        /Users/cad/go/pkg/mod/github.com/spf13/cobra@v1.0.0/command.go:831
github.com/spf13/cobra.(*Command).ExecuteC
        /Users/cad/go/pkg/mod/github.com/spf13/cobra@v1.0.0/command.go:950
github.com/spf13/cobra.(*Command).Execute
        /Users/cad/go/pkg/mod/github.com/spf13/cobra@v1.0.0/command.go:887
main.main
        /Users/cad/go/pkg/mod/github.com/volatiletech/sqlboiler/v4@v4.5.0/main.go:122
runtime.main
        /opt/homebrew/Cellar/go/1.16/libexec/src/runtime/proc.go:225
runtime.goexit
        /opt/homebrew/Cellar/go/1.16/libexec/src/runtime/asm_arm64.s:1130
unable to initialize tables
github.com/volatiletech/sqlboiler/v4/boilingcore.New
        /Users/cad/go/pkg/mod/github.com/volatiletech/sqlboiler/v4@v4.5.0/boilingcore/boilingcore.go:95
main.preRun
        /Users/cad/go/pkg/mod/github.com/volatiletech/sqlboiler/v4@v4.5.0/main.go:215
github.com/spf13/cobra.(*Command).execute
        /Users/cad/go/pkg/mod/github.com/spf13/cobra@v1.0.0/command.go:831
github.com/spf13/cobra.(*Command).ExecuteC
        /Users/cad/go/pkg/mod/github.com/spf13/cobra@v1.0.0/command.go:950
github.com/spf13/cobra.(*Command).Execute
        /Users/cad/go/pkg/mod/github.com/spf13/cobra@v1.0.0/command.go:887
main.main
        /Users/cad/go/pkg/mod/github.com/volatiletech/sqlboiler/v4@v4.5.0/main.go:122
runtime.main
        /opt/homebrew/Cellar/go/1.16/libexec/src/runtime/proc.go:225
runtime.goexit
        /opt/homebrew/Cellar/go/1.16/libexec/src/runtime/asm_arm64.s:1130
```

### Please provide a relevant database schema so we can replicate your issue (Provided you are comfortable sharing this)

```sql
                     Table "foo.bar"
   Column   |           Type           | Collation | Nullable |      Default       
------------+--------------------------+-----------+----------+--------------------
 id         | uuid                     |           | not null | uuid_generate_v4()
Indexes:
    "bar_pkey" PRIMARY KEY, btree (id)
```

### Further information. What did you do, what did you expect?

Generation of the models is expected. Instead sqlboiler fails with `unable to fetch table column info (bar): unable to scan for table bar: sql: Scan error on column index 1, name "column_type": converting NULL to string is unsupported
` error. 

P.s. This error doesn't occur on sqlboiler @ v4.4.0

**Comment by aarondl** [comment]
Interesting. I'm not actually able to reproduce this on 4.5.0. Every database I use with sqlboiler has this exact same pattern with one exception, your default value is a little strange.

The pg_crypto extension uuid isn't to be used anymore as a default value, as this built-in function is now supposed to be used:
https://www.postgresql.org/docs/13/functions-uuid.html

Have you made sure to recently update the `sqlboiler-psql` driver?

**Comment by cad** [comment]
My Go version is:

```bash
$ go version
go version go1.16.3 darwin/arm64
```

I've installed the `sqlboiler` and `sqlboiler-psql` binaries like this:

```bash
$ go install github.com/volatiletech/sqlboiler/v4@v4.5.0

$ go install github.com/volatiletech/sqlboiler/v4/drivers/sqlboiler-psql@v4.5.0
```

And I changed the default value to `gen_random_uuid()` and tried again from scratch and still getting the same error.

When I change the versions to `4.4.0` above, it just works.

**Comment by aarondl** [comment]
Versions:
```text
go version go1.16.5 linux/amd64
psql (PostgreSQL) 13.2 (Ubuntu 13.2-1.pgdg18.04+1)
SQLBoiler v4.5.0 (installed by copy pasting the above commands you ran, you can see below the time/date of installation, these are also the correct files that are being run because `which sqlboiler` gives me the right filepaths as output)
Jun  6 17:50 sqlboiler
Jun  6 17:49 sqlboiler-psql
```

Schema:
```sql
drop table if exists err_table;
create table err_table (
  id uuid not null primary key default gen_random_uuid()
);
```

config:
```toml
[psql]
  dbname = "issue"
  host   = "localhost"
  port   = 5432
  user   = "issue"
  pass   = "issue"
  blacklist = ["migrations"]
```

I simply can't reproduce this in any way. At this point I don't see any way around it other than you digging in pretty deep.

> unable to fetch table column info (bar): unable to scan for table bar: sql: Scan error on column index 1, name "column_type": converting NULL to string is unsupported

This error is coming from the following code: https://github.com/volatiletech/sqlboiler/blob/master/drivers/sqlboiler-psql/driver/psql.go#L308

The real question is how in the world is your `column_type` `NULL`? Are you maybe using a special `uuid` user defined type instead of the built in one somehow? You could attempt run the query here with the placeholder values replaced with your table and see what happens in your database, why is that `NULL`?:
https://github.com/volatiletech/sqlboiler/blob/master/drivers/sqlboiler-psql/driver/psql.go#L197

**Comment by cad** [comment]
Sorry for the time waste, I am no longer having this issue right now. 

Most probably it was something related to the Postgres.app or anything else, which may be solved on its own after upgrading to macOS 11.4 (which I did recently). In any case, it's now obvious that the problem was not rooted from sqlboiler end since I've been testing against the exact same sqlboiler release from the beginning. 

Thus I am closing this and you can as well mark this issue as "invalid".
